src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -81,7 +83,10 @@ public IEnumerable<string> GetManifestDirectories()
                 {
                     foreach (var workloadManifestDirectory in Directory.EnumerateDirectories(_manifestDirectories[0]))
                     {
-                        yield return workloadManifestDirectory;
+                        if (!_preview4ManifestIds.Any(prev4Manifest => workloadManifestDirectory.ToLowerInvariant().Contains(prev4Manifest.ToLowerInvariant())))
```

Comment:
Instead of just checking if the string contains the manifest ID, we should be parsing out the manifest ID from `workloadManifestDirectory` (I think we can do this just with `Path.GetFileName`), and comparing that manifest ID to the list of outdated manifest IDs.